icu4c/source/i18n/regexcmp.cpp

```diff
@@ -53,7 +53,7 @@ U_NAMESPACE_BEGIN
 //
 //------------------------------------------------------------------------------
 RegexCompile::RegexCompile(RegexPattern *rxp, UErrorCode &status) :
-   fParenStack(status), fSetStack(status), fSetOpStack(status)
```

Comment:
Since the `fSetStack` stack has a deleter function now, and calling `pop` doesn't delete the element, does this also mean that the 3 calls to `fSetStack.pop()` on lines 4625, 4632, and 4638 will need to explicitly call `delete` on the popped element(s)?

https://github.com/unicode-org/icu/blob/main/icu4c/source/i18n/regexcmp.cpp#L4628-L4641

Reply:
I think everything is OK.

For non-error cases, the logic was OK before, when fSetStack did not have a deleter, and is unchanged. Either way, ``fSetStack.pop()`` doesn't delete.  ``pop()`` can't delete - it's defined to return the popped item, which would be meaningless if it were already deleted.

At line 4616 ``rightOperand = (UnicodeSet *)fSetStack.peek();``
grabs the top of stack. rightOperand is subsequently deleted when needed.

The main reason for adding the deleter function to fSetStack was to make the error recovery code around pushing items onto the stack work correctly.  At line 4656, where push wants to adopt, including delete-on-error.

Reply:
Ah, I missed the `rightOperand` (which gets deleted in the 3 cases where `pop()` is called).
Thanks Andy!